node/resolvers/profile/index.ts

```diff
@@ -18,28 +18,37 @@ const TRUE = 'True'
 const FALSE = 'False'
 
 interface CheckUserAuthorizationParams {
-  identity: IdentityDataSource
-  storeUserAuthToken: IOContext['storeUserAuthToken']
+  context: Context
   email: string
 }
 
 const checkUserAuthorization = async ({
-  identity,
-  storeUserAuthToken,
+  context,
   email,
 }: CheckUserAuthorizationParams): Promise<boolean> => {
+  const {
+    vtex: { storeUserAuthToken, account },
+    dataSources: { identity },
+  } = context
+
   let validUser = !!storeUserAuthToken
-  let userTokenData: Partial<User> | null = { user: '' }
 
-  if (storeUserAuthToken) {
-    userTokenData = await identity.getUserWithToken(storeUserAuthToken)
-  }
+  const tokenUser = await identity.getUserWithToken(storeUserAuthToken ?? '')
+
+  const allowedAccount = await allowedAccounts(
+    context,
+    'account' in tokenUser ? tokenUser.account : ''
+  )
 
   validUser =
-    validUser && !!userTokenData && userTokenData?.user?.length === email.length
+    validUser &&
+    Boolean(tokenUser) &&
+    'id' in tokenUser &&
+    (!allowedAccount || tokenUser.account === account) &&
```

Comment:
If it is an `allowedAccount` we check if the accounts match, but if it isn't an `allowedAccount` we don't check it and then we make it a `validUser`? Is that right?

Reply:
Nope, good point, it's wrong. I'll fix it.